api/v1beta1/azuremanagedmachinepool_webhook.go

```diff
@@ -393,3 +401,38 @@ func (m *AzureManagedMachinePool) validateKubeletConfig() error {
 	}
 	return nil
 }
+
+// validateLinuxOSConfig enforces AKS API configuration for Linux OS custom configuration
+// See: https://learn.microsoft.com/en-us/azure/aks/custom-node-configuration#linux-os-custom-configuration for detailed information.
+func (m *AzureManagedMachinePool) validateLinuxOSConfig() error {
+	if m.Spec.LinuxOSConfig == nil {
+		return nil
+	}
+
+	if m.Spec.LinuxOSConfig.SwapFileSizeMB != nil {
+		if m.Spec.KubeletConfig == nil || m.Spec.KubeletConfig.FailSwapOn == nil || *m.Spec.KubeletConfig.FailSwapOn {
+			return field.Invalid(
+				field.NewPath("Spec", "KubeletConfig", "FailSwapOn"),
+				m.Spec.KubeletConfig.FailSwapOn,
+				"FailSwapOn must be set to false to enable swap file on nodes")
+		}
+	}
```

Comment:
This is something I found during my testing; `failSwapOn` must be set to false for `m.Spec.LinuxOSConfig.SwapFileSizeMB` to be accepted by Azure SDK